docs/DESIGN.md

```diff
@@ -0,0 +1,423 @@
+# Boulder flow diagrams
+
+Boulder is built in a decentralized way in order to enable parts to be deployed
```

Comment:
How about "Boulder is built out of multiple components that can be deployed in different security contexts?" I think "decentralized" here might lead to some minor confusion.